src/main/java/org/jenkinsci/plugins/gitlab/GitlabBuilds.java

```diff
@@ -23,18 +33,132 @@ public GitlabBuilds(GitlabBuildTrigger trigger, GitlabRepository repository) {
         this.repository = repository;
     }
 
-    public String build(GitlabMergeRequestWrapper mergeRequest, Map<String, String> customParameters) {
-        GitlabCause cause = new GitlabCause(mergeRequest.getId(), mergeRequest.getIid(),
-                mergeRequest.getSourceName(), mergeRequest.getSourceRepository(),
-                mergeRequest.getSourceBranch(), mergeRequest.getTargetBranch(), customParameters, mergeRequest.getDescription());
-
-        QueueTaskFuture<?> build = trigger.startJob(cause);
-        if (build == null) {
-            LOGGER.log(Level.SEVERE, "Job failed to start.");
+    public String build(GitlabCause cause, Map<String, String> customParameters) throws IOException {
+    	
+    	boolean shouldRun = true;
+    	GitlabAPI api = trigger.getBuilder().getGitlab().get();
+    	GitlabProject project = api.getProject(cause.getSourceProjectId());
+    	GitlabMergeRequest mergeRequest = api.getMergeRequest(project, cause.getMergeRequestId());
```

Comment:
I was thinking about pushing some of these API calls back up to their caller, because the project and mergeRequest are already available in the GitlabMergeRequestWrapper; so the API calls would just need to be done in the GitlabWebhook class and passed in.